File: Data Structures & Algorithms/contains-duplicate-ii/submission-0.go
func containsNearbyDuplicate(nums []int, k int) bool {
	seen := make(map[int]int)

	for i, num := range nums {
		if val, ok := seen[num]; ok {
			if i - val <= k {
				return true
			}
		}
		seen[num] = i
	}

	return false
}

File: Data Structures & Algorithms/missing-number/submission-0.py
class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        seen = set(nums)
        n = len(nums)
        
        # Check every number that *should* be there
        for i in range(n + 1):
            if i not in seen:
                return i
                
        return -1